==72891== Helgrind, a thread error detector
==72891== Copyright (C) 2007-2017, and GNU GPL'd, by OpenWorks LLP et al.
==72891== Using Valgrind-3.21.0.GIT-lbmacos and LibVEX; rerun with -h for copyright info
==72891== Command: ./philo 1 250 100 100
==72891== Parent PID: 33999
==72891== 
--72891-- run: /usr/bin/dsymutil "./philo"
==72891== ---Thread-Announcement------------------------------------------
==72891== 
==72891== Thread #2 was created
==72891==    at 0x1005CEDB6: __bsdthread_create (in /usr/lib/system/libsystem_kernel.dylib)
==72891==    by 0x100630DF3: _pthread_create (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1000018DC: thinking_currents (philo.c:22)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== ---Thread-Announcement------------------------------------------
==72891== 
==72891== Thread #1 is the program's root thread
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x70000DAC6034 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E1A0: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #1
==72891== Locks held: none
==72891==    at 0x1006316FA: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x70000dac6034 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x100635270 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E1AB: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #1
==72891== Locks held: none
==72891==    at 0x100630D6C: _pthread_create (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001909: thinking_currents (philo.c:26)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x100635270 is in the Data segment of /usr/lib/system/libsystem_pthread.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x100635270 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E1B7: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #1
==72891== Locks held: none
==72891==    at 0x100630D6C: _pthread_create (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001909: thinking_currents (philo.c:26)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x100635270 is in the Data segment of /usr/lib/system/libsystem_pthread.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x70000DAC6020 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E1BD: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #1
==72891== Locks held: none
==72891==    at 0x1006316FE: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x70000dac6020 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 2 at 0x1049155EC by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBCD: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 2 by thread #1
==72891== Locks held: none
==72891==    at 0x1002E15A0: dyld_stub_binder (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x10000806F: ??? (in ./philo)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x1049155ec is on thread #1's stack
==72891==  in frame #1, created by _pthread_join (???:)
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 1 at 0x1049155EE by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBCD: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 1 by thread #1
==72891== Locks held: none
==72891==    at 0x1002E15A0: dyld_stub_binder (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x10000806F: ??? (in ./philo)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x1049155ee is on thread #1's stack
==72891==  in frame #1, created by _pthread_join (???:)
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x70000DAC6030 by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBD6: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous read of size 4 by thread #1
==72891== Locks held: none
==72891==    at 0x100631533: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x70000dac6030 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x1049155E0 by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBD9: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #1
==72891== Locks held: none
==72891==    at 0x1002E15A4: dyld_stub_binder (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x10000806F: ??? (in ./philo)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x1049155e0 is on thread #1's stack
==72891==  in frame #1, created by _pthread_join (???:)
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x70000DAC6010 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E1E5: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #1
==72891== Locks held: none
==72891==    at 0x100630D69: _pthread_create (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001909: thinking_currents (philo.c:26)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x70000dac6010 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 8 at 0x70000DB49018 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E208: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000db49018 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 8 at 0x1000DBDD0 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E210: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891== 
==72891== This conflicts with a previous read of size 8 by thread #1
==72891== Locks held: none
==72891==    at 0x1006314E1: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891==  Address 0x1000dbdd0 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x70000DAC6034 by thread #1
==72891== Locks held: none
==72891==    at 0x100631659: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E1A0: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000dac6034 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 8 at 0x70000DAC6020 by thread #1
==72891== Locks held: none
==72891==    at 0x100631736: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous read of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E1BD: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000dac6020 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x70000DAC6030 by thread #1
==72891== Locks held: none
==72891==    at 0x10063173E: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBD6: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000dac6030 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x70000DAC60C0 by thread #1
==72891== Locks held: none
==72891==    at 0x10062FC7A: _pthread_deallocate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100631776: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E101: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000dac60c0 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x70000DAC60C8 by thread #1
==72891== Locks held: none
==72891==    at 0x10062FC81: _pthread_deallocate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100631776: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001946: thinking_currents (philo.c:30)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E10E: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000dac60c8 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 8 at 0x1049155E0 by thread #1
==72891== Locks held: none
==72891==    at 0x100631485: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous read of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBD9: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x1049155e0 is on thread #1's stack
==72891==  in frame #0, created by _pthread_join (???:)
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 2 at 0x1049155EC by thread #1
==72891== Locks held: none
==72891==    at 0x10063148B: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous read of size 2 by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBCD: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x1049155ec is on thread #1's stack
==72891==  in frame #0, created by _pthread_join (???:)
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 1 at 0x1049155EE by thread #1
==72891== Locks held: none
==72891==    at 0x10063148B: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous read of size 1 by thread #2
==72891== Locks held: none
==72891==    at 0x10062FBCD: _pthread_joiner_prepost_wake (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x1049155ee is on thread #1's stack
==72891==  in frame #0, created by _pthread_join (???:)
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x1000DBDD0 by thread #1
==72891== Locks held: none
==72891==    at 0x1006314E1: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E210: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x1000dbdd0 is in a rw- anonymous segment
==72891== 
==72891== ---Thread-Announcement------------------------------------------
==72891== 
==72891== Thread #3 was created
==72891==    at 0x1005CEDB6: __bsdthread_create (in /usr/lib/system/libsystem_kernel.dylib)
==72891==    by 0x100630DF3: _pthread_create (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100001909: thinking_currents (philo.c:26)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x70000DB49030 by thread #1
==72891== Locks held: none
==72891==    at 0x100631533: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #3
==72891== Locks held: none
==72891==    at 0x10062E200: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000db49030 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x70000DB49034 by thread #1
==72891== Locks held: none
==72891==    at 0x1006316E1: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #3
==72891== Locks held: none
==72891==    at 0x10062E1A0: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000db49034 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x70000DB49018 by thread #1
==72891== Locks held: none
==72891==    at 0x1006317D5: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E208: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000db49018 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 8 at 0x1000DBDD0 by thread #1
==72891== Locks held: none
==72891==    at 0x1006317EF: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062E210: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x1000dbdd0 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 1 at 0x70000DB49031 by thread #1
==72891== Locks held: none
==72891==    at 0x1006317F2: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #3
==72891== Locks held: none
==72891==    at 0x10062E200: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000db49031 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x70000DB490C0 by thread #1
==72891== Locks held: none
==72891==    at 0x10062FC7A: _pthread_deallocate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100631776: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #3
==72891== Locks held: none
==72891==    at 0x10062E101: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000db490c0 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x70000DB490C8 by thread #1
==72891== Locks held: none
==72891==    at 0x10062FC81: _pthread_deallocate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100631776: _pthread_join (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10000196B: thinking_currents (philo.c:33)
==72891==    by 0x100001A0B: main (philo.c:47)
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #3
==72891== Locks held: none
==72891==    at 0x10062E10E: _pthread_terminate (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062E09B: _pthread_terminate_invoke (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10063053A: _pthread_exit (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100630113: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x70000db490c8 is in a rw- anonymous segment
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 1 at 0x10082C118 by thread #1
==72891== Locks held: none
==72891==    at 0x10062BFB5: pthread_mutex_destroy (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100002434: free_all (utils.c:115)
==72891==    by 0x100001A17: main (philo.c:48)
==72891== 
==72891== This conflicts with a previous write of size 1 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B618: pthread_mutex_lock (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==    by 0x100630108: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x10082c118 is 24 bytes inside a block of size 64 alloc'd
==72891==    at 0x100124FA5: malloc (in /Users/pcazac/.brew/Cellar/valgrind/HEAD-0d19de6/libexec/valgrind/vgpreload_helgrind-amd64-darwin.so)
==72891==    by 0x1000021CC: ft_calloc (utils.c:72)
==72891==    by 0x100001EA8: fill_struct (struct_utils.c:83)
==72891==    by 0x1000019E4: main (philo.c:44)
==72891==  Block was alloc'd by thread #1
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 1 at 0x10082C100 by thread #1
==72891== Locks held: none
==72891==    at 0x10062BFE7: pthread_mutex_destroy (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100002434: free_all (utils.c:115)
==72891==    by 0x100001A17: main (philo.c:48)
==72891== 
==72891== This conflicts with a previous read of size 1 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B5AD: pthread_mutex_lock (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==    by 0x100630108: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x10082c100 is 0 bytes inside a block of size 64 alloc'd
==72891==    at 0x100124FA5: malloc (in /Users/pcazac/.brew/Cellar/valgrind/HEAD-0d19de6/libexec/valgrind/vgpreload_helgrind-amd64-darwin.so)
==72891==    by 0x1000021CC: ft_calloc (utils.c:72)
==72891==    by 0x100001EA8: fill_struct (struct_utils.c:83)
==72891==    by 0x1000019E4: main (philo.c:44)
==72891==  Block was alloc'd by thread #1
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 1 at 0x10082C098 by thread #1
==72891== Locks held: none
==72891==    at 0x10062BFB5: pthread_mutex_destroy (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100002454: free_all (utils.c:117)
==72891==    by 0x100001A17: main (philo.c:48)
==72891== 
==72891== This conflicts with a previous write of size 1 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B618: pthread_mutex_lock (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100003044: think (philo_study.c:109)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==    by 0x100630108: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x10082c098 is 24 bytes inside a block of size 64 alloc'd
==72891==    at 0x100124FA5: malloc (in /Users/pcazac/.brew/Cellar/valgrind/HEAD-0d19de6/libexec/valgrind/vgpreload_helgrind-amd64-darwin.so)
==72891==    by 0x1000021CC: ft_calloc (utils.c:72)
==72891==    by 0x100001E41: fill_struct (struct_utils.c:75)
==72891==    by 0x1000019E4: main (philo.c:44)
==72891==  Block was alloc'd by thread #1
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 1 at 0x10082C080 by thread #1
==72891== Locks held: none
==72891==    at 0x10062BFE7: pthread_mutex_destroy (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100002454: free_all (utils.c:117)
==72891==    by 0x100001A17: main (philo.c:48)
==72891== 
==72891== This conflicts with a previous read of size 1 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B5AD: pthread_mutex_lock (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100003044: think (philo_study.c:109)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==    by 0x100630108: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062BB8A: thread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x10082c080 is 0 bytes inside a block of size 64 alloc'd
==72891==    at 0x100124FA5: malloc (in /Users/pcazac/.brew/Cellar/valgrind/HEAD-0d19de6/libexec/valgrind/vgpreload_helgrind-amd64-darwin.so)
==72891==    by 0x1000021CC: ft_calloc (utils.c:72)
==72891==    by 0x100001E41: fill_struct (struct_utils.c:75)
==72891==    by 0x1000019E4: main (philo.c:44)
==72891==  Block was alloc'd by thread #1
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x100415118 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B5AD: pthread_mutex_lock (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B813: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415118 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x100415124 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B667: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B6E2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415124 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x100415138 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B86F: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B7C5: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415138 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x100415130 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B894: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B7C2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415130 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 8 at 0x100415130 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B915: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B7C2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415130 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x100415124 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B94F: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B6E2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415124 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x100415380 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3070: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C6381: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415380 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 2 at 0x100415378 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3083: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 2 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C6395: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415378 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x100415368 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3093: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C637E: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415368 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x100415368 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3099: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C637E: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415368 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x100415374 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C30AC: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E20A: std::__1::__stdoutbuf<char>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1007844E8: std::__1::basic_ostream<char, std::__1::char_traits<char> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D889: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x1003D9C6E: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==    by 0x100630108: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x100415374 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x100415118 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B5AD: pthread_mutex_lock (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B813: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415118 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x100415124 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B667: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B6E2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415124 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x100415138 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B86F: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B7C5: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415138 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x100415130 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B894: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B7C2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415130 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 8 at 0x100415130 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B915: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B7C2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415130 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x100415124 by thread #1
==72891== Locks held: none
==72891==    at 0x10062B94F: _pthread_mutex_firstfit_lock_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C3013: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x10062B6E2: _pthread_mutex_check_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x10062B658: _pthread_mutex_lock_init_slow (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003C2A69: flockfile (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB297: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415124 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x100415380 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3070: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C6381: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415380 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 2 at 0x100415378 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3083: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 2 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C6395: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415378 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 4 at 0x100415368 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3093: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C637E: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415368 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x100415368 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C3099: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C637E: __smakebuf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003D9C4D: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==  Address 0x100415368 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during write of size 4 at 0x100415374 by thread #1
==72891== Locks held: none
==72891==    at 0x1003C30AC: __sflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C301B: fflush (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10078E536: std::__1::__stdoutbuf<wchar_t>::sync() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x100786AE4: std::__1::basic_ostream<wchar_t, std::__1::char_traits<wchar_t> >::flush() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x10078D895: std::__1::DoIOSInit::~DoIOSInit() (in /usr/lib/libc++.1.dylib)
==72891==    by 0x1003E413B: __cxa_finalize_ranges (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003E4411: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891== 
==72891== This conflicts with a previous write of size 4 by thread #2
==72891== Locks held: none
==72891==    at 0x1003D9C6E: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10000201B: safe_print (utils.c:20)
==72891==    by 0x100003061: think (philo_study.c:110)
==72891==    by 0x1000025BE: existential_crisis (philo_work.c:39)
==72891==    by 0x100630108: _pthread_start (in /usr/lib/system/libsystem_pthread.dylib)
==72891==  Address 0x100415374 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== ----------------------------------------------------------------
==72891== 
==72891== Possible data race during read of size 8 at 0x100414E50 by thread #1
==72891== Locks held: none
==72891==    at 0x1003E4412: exit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1002E2CCF: start (in /usr/lib/system/libdyld.dylib)
==72891==    by 0x4: ???
==72891==    by 0x104915832: ???
==72891==    by 0x10491583A: ???
==72891==    by 0x10491583C: ???
==72891==    by 0x104915840: ???
==72891==    by 0x104915844: ???
==72891== 
==72891== This conflicts with a previous write of size 8 by thread #2
==72891== Locks held: none
==72891==    at 0x1003C3C3E: __sinit (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x10062C7E4: __pthread_once_handler (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x100617EC6: _os_once_callout (in /usr/lib/system/libsystem_platform.dylib)
==72891==    by 0x10062C792: pthread_once (in /usr/lib/system/libsystem_pthread.dylib)
==72891==    by 0x1003D9BCE: __swsetup (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003F294E: __v2printf (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003CB2B1: vfprintf_l (in /usr/lib/system/libsystem_c.dylib)
==72891==    by 0x1003C941D: printf (in /usr/lib/system/libsystem_c.dylib)
==72891==  Address 0x100414e50 is in the Data segment of /usr/lib/system/libsystem_c.dylib
==72891== 
==72891== 
==72891== Use --history-level=approx or =none to gain increased speed, at
==72891== the cost of reduced accuracy of conflicting-access information
==72891== For lists of detected and suppressed errors, rerun with: -s
==72891== ERROR SUMMARY: 90 errors from 54 contexts (suppressed: 0 from 0)
